students/urtnowski_daniel/lesson_01_basics/Snakify_L2.py

```diff
@@ -0,0 +1,148 @@
+
+
+def fractional_part():
+    """
+    This function prints the fractional part of a given real number
+    """
+    print("Exercise 4: Fractional part")
+    number = float(input("Enter a number: "))
+
+    result = number - int(number)
+    delta = 0.0000000001
+    if fabs(result) < delta:
+        result = 0
+    else:
+        result = round(result, 10)
+
+    print("Fractional part of the number is: ", result)
+
+
+def first_digit_after_decimal_point():
+    """
+    This function prints the first digit to the right of the decimal point of a given positive real number
+    """
+    print("Exercise 5: First digit after decimal point")
+    number = float(input("Enter a number: "))
+
+    result = int(number * 10) % 10
+    print("The result is: ", result)
+
+
+def car_route():
+    """
+    This function prints how many days it will take to cover a route of a given length with given distance
+    a car can cover per day.
+    """
+    print("Exercise 6: Car route")
+    efficiency = int(input("Enter a number of kilometers the car can cover per day: "))
+    route_len = int(input("Enter a number of kilometers in the route: "))
+
+    result = ceil(route_len / efficiency)
+    print("The result is: ", result, " day(s)")
+
+
+def digital_clock():
+    """
+    This function reads integer number with minutes passed since midnight and prints how many hours and minutes
+    are displayed on the 24h digital clock
+    """
+    print("Exercise 7: Digital clock")
+    number = int(input("Enter a number of minutes passed since midnight: "))
+
+    hours = number // 60
+    minutes = number % 60
+    print("The actual time is: {}:{}".format(hours, minutes))
+
+
+def total_cost():
+    """
+    This function reads a cost of a cupcake in dollars and cents and prints the total cost of given number of cupcakes
+    """
+    print("Exercise 8: Total cost")
+    dollars_per_cupcake = int(input("Enter dollars part of cupcake price: "))
+    cents_per_cupcake = int(input("Enter cents part of cupcake price: "))
+    number_of_cupcakes = int(input("Enter number of cupcakes: "))
+
+    cupcake_cost_in_cents = (dollars_per_cupcake * 100) + cents_per_cupcake
+    total_cost_in_cents = cupcake_cost_in_cents * number_of_cupcakes
+
+    print("The total cost is: {} dollars {} cents".format(total_cost_in_cents // 100, total_cost_in_cents % 100))
+
+
+def clock_face_1():
+    """
+    This function reads how many hours, minutes and seconds are passed since the midnight and prints the angle hour
+    hand on the clock face right now
+    """
+    print("Exercise 9: Clock face - 1")
+    hours = int(input("Enter the hours number: "))
+    minutes = int(input("Enter the minutes number: "))
+    seconds = int(input("Enter the seconds number: "))
+
+    time_in_seconds = seconds + ((minutes + (hours * 60)) * 60)
+    seconds_per_day = 60 * 60 * 12
+    degrees_per_day = 360
+    angle = (time_in_seconds / seconds_per_day) * degrees_per_day
```

Comment:
1. Traktujesz seconds_per_day i degrees_per_day jako stałe. W pythonie język nie obsługuje definiowania stałych, więc pattern jest taki, że robi się zmienną i się jej nie rusza. Żeby pamiętać, że się nie rusza należy pilnować konwencji nazewniczej. PEP8 ma zalecenia co do nazewnictwa stałych https://www.python.org/dev/peps/pep-0008/#constants Zalecam poprawkę
2. W rozdziale https://www.python.org/dev/peps/pep-0008/#imports piszą, że stałe powinny się znaleźć przed właściwym kodem programu czyli przed inputami. Jak wywalisz stąd stałe to funkcja się skróci, co też jest na plus (krótszy kod i mniejszy stos). Zalecam poprawkę zgodnie z:

> Imports are always put at the top of the file, just after any module comments and docstrings, and before module globals and constants.